spec/system/candidate_signs_up_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+describe 'A candidate signs up to Apply service' do
+  context 'when a new candidate tries to sign in' do
+    before do
+      visit '/'
+      click_on t('application_form.begin_button')
+      fill_in t('authentication.sign_up.email_address.label'), with: 'april@pawnee.com'
+      click_on t('authentication.sign_up.button')
+    end
+
+    it 'can allow a candidate to submit their email' do
+      expect(page).to have_content t('authentication.check_your_email')
+    end
+
+    it 'can sign in a candidate using the magic link' do
+      open_email('april@pawnee.com')
+      current_email.find_css('a').first.click
+
+      expect(page).to have_content 'april@pawnee.com'
+    end
+  end
+
+  context 'when a candidate is not signed in' do
+    let(:redirect_url) { root_url }
+
+    it 'can stop a candidate from viewing the personal details page' do
+      visit '/personal-details/new'
```

Comment:
Let's not do try to hit all the cases in system specs — we'll eventually make a mistake and it will be as good as not having specs at all. Can this behaviour be tested some other way?

Reply:
Moved these tests into separate tests for each controller. 
These tests only check that the user is redirected if there is no session.